api/Dockerfile

```diff
@@ -23,21 +24,29 @@ RUN adduser \
 WORKDIR /app
 
 # First we handle fetching and building our dependencies
-# We need a stub entry point for that
+# This rarely changes so is done once and for all
+# Cargo needs a stub entry point otherwise it fails
 COPY Cargo.toml Cargo.lock ./
 COPY ./src/stub.rs ./src/main.rs
 RUN cargo fetch --target x86_64-unknown-linux-musl
 RUN cargo build --offline --target x86_64-unknown-linux-musl --release
 
+# Then we copy every files, and detect linting / formating errors
+# This is only detection, fixing the errors should be done outside docker
+# We handle this section in offline mode to avoid unecessary and terribly long crates index update
 COPY ./ .
 ENV SQLX_OFFLINE=true
-RUN cargo clippy --target x86_64-unknown-linux-musl --tests --examples --offline -- -D warnings
 RUN cargo fmt --all -- --check
+RUN cargo clippy --target x86_64-unknown-linux-musl --tests --examples --offline -- -D warnings --no-deps
+RUN cargo test tests --target x86_64-unknown-linux-musl
```

Comment:
This does not run the integration tests in `/tests`

Reply:
Indeed, it is now fixed, thanks.